### PR TITLE
New: add more name IDs and make sure the correct settings are fontRev…

### PR DESCRIPTION
**Pull request by yisibl**
1. The sfntNames now support more fields:
  * designer
  * designerurl
  * license
  * licenseurl

2. `fontRevision` It is no longer always `1.0`.
Based on the recommendations[1] in this document, we can now support better version number settings. It can be displayed correctly in the font editing software.

e.g. `Version 5.678 beta3`

FontLab VI

TODO: add test cases.

[1] https://silnrsi.github.io/FDBP/en-US/Versioning.html

### CODE REVIEW

**Comment by yisibl** [comment]
@puzrin please take a look?

**Comment by puzrin** [comment]
I'd say, project tests should be reorganized with use of `opentype.js` instead of binary compare. Because everything flies to nightmare :).

Without normal testing it's very difficult to move forward.

File: index.js [hunk 1]
```diff
@@ -40,10 +44,38 @@ function svg2ttf(svgString, options) {
...
+  // Like SourceSansPro's Version 1.095;hotconv 1.0.109;makeotfexe 2.5.65596
+  // or 10.006 beta1
+  versionString = 'Version ' + fontVersion[2];
+
+  // We must update the fontRevision field and ensure that a specific string is
+  // supported. See also: https://silnrsi.github.io/FDBP/en-US/Versioning.html
+  font.revision = fontVersion[3]; // Only number
+
   font.sfntNames.push({ id: 5, value: versionString }); // version ID for TTF name table
   font.sfntNames.push({ id: 6, value: (options.fullname || svgFont.id).replace(/[\s\(\)\[\]<>%\/]/g, '').substr(0, 62) }); // Postscript name for the font, required for OSX Font Book
```
Comment (yisibl) [inline index.js:58]: In name id 5, the minor version should always be three digits and should refer to the UFO specification（openTypeNameVersion）: https://unifiedfontobject.org/versions/ufo3/fontinfo.plist/#opentype-name-table-fields，I can also consider adding openTypeNameUniqueID

Comment (yisibl) [reply index.js:58]: In the implementation of ufo2ft, the default version is always 0.000. Use Python's `.zfill(3)` to fill 0 in front of the minor version, similar to `.padStart(3, '0')` in JS.

https://github.com/googlefonts/ufo2ft/blob/742b3a54408dbba99f24a9d21e32b0c499ca6ffe/Lib/ufo2ft/fontInfoData.py#L169-L175

**Comment by puzrin** [comment]
IMO this is not enough flexible and should be reworked.

1. Split for 2 parts: pure string add and version magic.
2. Api for string add clutter options significantly. IMO this can be added via more abstract interface, list of `[ data, id, platformID, encodingID, languageID ]`. That will allow to add any vendor string user can imagine.
3. I don't like existing magic with version. It can exists, to extract 2 major numbers, when possible, but `--version-num` also required to define header value explicit.

**Comment by puzrin** [comment]
@yisibl here are 2 PRs and 1 issue i'd like to close/replace: #86, #102, #105.

Reason: topics were mutated multiple time, and tend to be "endless".

My last opinion https://github.com/fontello/svg2ttf/pull/86#issuecomment-846154816 is to create 2 new issues if that's still actual. AFAIK, all other critical things have been landed to master. Let me know what you think.

**Comment by yisibl** [comment]
I'm very sorry for the late reply. When I have time, I will investigate how the APIs of other libraries are handled. You can close these issues first. Thanks!

**Comment by puzrin** [comment]
Closed in favor of #111 & #112

**Pull request closed without merging**